pkg/jmxfetch/utils.go

```diff
@@ -0,0 +1,43 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build jmx
+
+package jmxfetch
+
+import "github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+
+// GetJSONSerializableMap returns a JSON serializable map from a raw map
+func GetJSONSerializableMap(m interface{}) interface{} {
+	switch x := m.(type) {
+	// unbelievably I cannot collapse this into the next (identical) case
+	case map[interface{}]interface{}:
+		j := integration.JSONMap{}
+		for k, v := range x {
+			j[k.(string)] = GetJSONSerializableMap(v)
```

Comment:
are we sure that in every cases `k` is a string ?

Reply:
I'm not changing anything in this PR, just moving code around